Kudu.Core/Deployment/FetchDeploymentManager.cs

```diff
@@ -105,11 +106,14 @@ public class FetchDeploymentManager : IFetchDeploymentManager
                 {
                     if (PostDeploymentHelper.IsAutoSwapOngoing())
                     {
-                        return FetchDeploymentRequestResult.ConflictAutoSwapOngoing;
+                        return new FetchDeploymentRequestResult(FetchDeploymentRequestResultStatus.ConflictAutoSwapOngoing);
                     }
 
                     await PerformDeployment(deployInfo);
-                    return FetchDeploymentRequestResult.RanSynchronously;
+                    var result = _deploymentManager.GetResults().FirstOrDefault();
```

Comment:
GetResults() is not a good idea here (it hits the filesystem to enumerate all the existing results, does locking stuff, and has the side effect of purging old deployment records, none of which I want to incur here) but unfortunately there's no other option the way things are factored now.

DeploymentManager.DeployAsync() currently doesn't return anything. Given the functionality it would make sense to return the DeployResult, and in turn that could be returned up through PerformDeployment() and checked here. Please give that a go; if it's not looking great let me know and I'll take a swing at it or something similar.

Reply:
Now using `GetResult(id)` to return the DeployResult from `DeployAsync`? Does that solve the lockingissues?